experimental/InvariantTheory/primary_invariants.jl

```diff
@@ -347,12 +358,21 @@ julia> primary_invariants(IR)
 function primary_invariants(IR::InvRing, algo::Symbol = :optimal_hsop)
   if !isdefined(IR, :primary)
     if algo == :optimal_hsop
-      IR.primary = primary_invariants_via_optimal_hsop(IR)
+      IR.primary = _primary_invariants_via_optimal_hsop(IR)
     elseif algo == :successive_algo
-      IR.primary = primary_invariants_via_successive_algo(IR)
+      IR.primary = _primary_invariants_via_successive_algo(IR)
```

Comment:
Is this variant tested?

Reply:
I'm stupid. There are tests for it, but the caching gets into the way...

Reply:
Now I get some funny Singular output in the modular case:
```
// ** int division with `/`: use `div` instead in line >>      out=out+(i/p^(j-1))*a^(j-1);<<
```
(repeats ~800 times)

Reply:
Let's ignore it for the moment.

https://github.com/oscar-system/Singular.jl/issues/558